src/pages/Editor.vue

```diff
@@ -602,35 +666,169 @@ export default {
+      // set up the dialog properties when user tries to add an item
+      // at an invalid time
+      this.dialogTitle = "Cannot add a new question here";
+      this.dialogDescription = "Questions should be at least 2 seconds apart. Please choose a different time for the question";
+      this.dialogConfirmButtonConfig = {
+        enabled: true,
+        text: "Got it!",
+        class: "bg-primary-button hover:bg-primary-button-hover focus:outline-none focus:ring-0",
+      };
+      this.dialogCancelButtonConfig = {
+        enabled: false,
+        text: "",
+        class: ""
+      };
+
+      // carry out the closeDialog action when dialog is closed
+      this.dialogAction = "closeDialog"
+      // show the dialogue
+      this.showDialogBox = true;
+    },
+    dialogPublish() {
+      this.dialogTitle = this.publishInProgressDialogTitle;
+      // publish the plio or its changes
+      this.publishPlio();
+    },
+    dialogDeleteItem() {
+      this.dialogTitle = this.deleteItemInProgressDialogTitle;
+      // delete the selected item after user confirms
+      this.deleteSelectedItem();
+    },
+    getTimestampForNewItem() {
+      // loop through itemTimestamps to check if the time where the user
+      // is trying to add an item is valid or not
+      // using for loop instead of forEach as forEach was running async
+
+      // returning -1 to signify that the time chosen by the user is not valid
+      for (let index = 0; index < this.itemTimestamps.length; index++) {
+        var val = this.itemTimestamps[index]
+        if (val == this.currentTimestamp) return -1
+        if ( this.currentTimestamp <= (val + 2) && this.currentTimestamp >= (val - 2) )
+          return -1
+      }
+      return this.currentTimestamp
+    },
+    getItemTypeForNewItem() {
+      // hardcoded for now
+      return "question"
+    },
+    getMetadataForNewItem() {
+      // defaults to "default" right now
+      var metadata = {}
+      metadata['source'] = {}
+      metadata['source']['name'] = "default"
+      return metadata
+    },
+    getDetailsForNewItem() {
+      // barebones question structure
+      var details = {}
+      details['correct_answer'] = 0
+      details['text'] = ""
+      details['type'] = "mcq_single_answer"
+      details['options'] = ["", ""]
+      return details
+    },
+    addNewItem() {
+      this.$refs.playerObj.player.pause();
+      // get all the data needed to create a new item
+      var newItem = {}
+      var newTimestamp = this.getTimestampForNewItem()
+      var newType = this.getItemTypeForNewItem()
+      var newMetadata = this.getMetadataForNewItem()
+      var newItemDetails = this.getDetailsForNewItem()
+
+      if(newTimestamp == -1){
+        this.showCannotAddItemDialog()
+        return
+      }
+
+      // create the newItem object
+      newItem['time'] = newTimestamp
+      newItem['type'] = newType
+      newItem['metadata'] = newMetadata
+      newItem['details'] = newItemDetails
```

Comment:
these local variables don't need to be created no? can directly assign value from the function output?

Reply:
Makes sense. Just keeping the variable newTimestamp as I need to check and return if it's -1